core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -135,6 +137,19 @@ public <R> R doWithTopicOps(ThrowableFunction<TopicCatalog, R> fn) throws Except
           });
     }
 
+    public <R> R doWithPartitionOps(
+        NameIdentifier tableIdent, ThrowableFunction<SupportsPartitions, R> fn) throws Exception {
+      return classLoader.withClassLoader(
+          cl -> {
+            Preconditions.checkArgument(
+                asTables() != null, "Catalog does not support table operations");
+            Table table = asTables().loadTable(tableIdent);
```

Comment:
Can we load table first and then as a parameter to this method? So we would not load the table each time.

Reply:
Yes, but there is a trade-off:

1. load table first and then as a parameter to this method: load table once, switch thread class loader 4 times(2 for load table, 2 for partition ops)
2. current implementation: load table once (Since each request will only perform one partition operation currently), switch thread class loader 2 times

Therefore, I think it's fine for the current implementation, WDYT?

Reply:
The current implementation is fine as is, and we can change it after the full test is completed.